lib/DataStore.js

```diff
@@ -198,7 +203,7 @@ DataStore.prototype.getRoomIdsFromConfig = function() {
 DataStore.prototype.removeConfigMappings = function() {
     log.debug("removeConfigMappings ");
     return this._roomStore.removeEntriesByLinkData({
-        from_config: true
+        origin: 'config'
```

Comment:
This breaks backwards compatibility.

Some existing databases will have `from_config: true` in the `.db` file. When the new code runs, the old config mappings will not be removed because of this.

Port across to the new style by also removing entries where `from_config: true`.